src/util/time.rs

```diff
@@ -171,16 +191,114 @@ mod inner {
     use libc;
 
     pub fn raw_now() -> Timespec {
+        get_time(libc::CLOCK_MONOTONIC_RAW as _)
+    }
+
+    pub fn coarse_now() -> Timespec {
+        get_time(libc::CLOCK_MONOTONIC_COARSE as _)
+    }
+
+    fn get_time(clock: libc::clock_t) -> Timespec {
         let mut t = libc::timespec {
             tv_sec: 0,
             tv_nsec: 0,
         };
-        let res = unsafe { libc::clock_gettime(libc::CLOCK_MONOTONIC_RAW, &mut t) };
-        if res != 0 {
+        let errno = unsafe { libc::clock_gettime(clock as _, &mut t) };
+        if errno != 0 {
             panic!("failed to get monotonic raw locktime, err {}",
                    io::Error::last_os_error());
         }
-        Timespec::new(t.tv_sec, t.tv_nsec as i32)
+        Timespec::new(t.tv_sec, t.tv_nsec as _)
+    }
+}
+
+/// A measurement of a monotonically increasing clock.
+/// It's similar and meat to replace `std::time::Instant`,
+/// for providing extra features.
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug)]
+pub enum Instant {
+    Monotonic(Timespec),
+    MonotonicCoarse(Timespec),
+}
+
+impl Instant {
+    pub fn now() -> Instant {
+        Instant::Monotonic(raw_now())
+    }
+
+    pub fn now_coarse() -> Instant {
+        Instant::MonotonicCoarse(coarse_now())
+    }
+
+    pub fn elapsed(&self) -> Duration {
+        match *self {
+            Instant::Monotonic(t) => {
+                let now = raw_now();
+                elapsed_duration(now, t)
+            }
+            Instant::MonotonicCoarse(t) => {
+                let now = coarse_now();
+                elapsed_duration(now, t)
+            }
+        }
+    }
+
+    pub fn duration_since(&self, earlier: Instant) -> Duration {
+        let later = match *self {
+            Instant::Monotonic(t) |
+            Instant::MonotonicCoarse(t) => t,
+        };
+        let earlier = match earlier {
+            Instant::Monotonic(t) |
+            Instant::MonotonicCoarse(t) => t,
+        };
+        elapsed_duration(later, earlier)
+    }
+}
+
+impl Add<Duration> for Instant {
```

Comment:
It is **necessary**. https://github.com/pingcap/tikv/blob/13a8b89a94c0bda9f812df003d0d3a4455801034/src/coprocessor/endpoint.rs#L131-L134